This patch from aorlinsk fixes an issue with order in options passed to tar

  http://openwrt.org/forum/viewtopic.php?pid=8332#p8332


Index: ipkg-utils-1.7/ipkg-build
===================================================================
--- ipkg-utils-1.7.orig/ipkg-build	2007-06-30 14:19:16.519409000 +0200
+++ ipkg-utils-1.7/ipkg-build	2007-06-30 14:20:12.722921500 +0200
@@ -184,7 +184,7 @@
 	g ) group=$OPTARG
 	    ogargs="$ogargs --group=$group"
 	    ;;
-        c ) outer=tar
+        c ) outer=$TAR
             ;;
         C ) noclean=1
             ;;
@@ -243,8 +243,8 @@
 mkdir $tmp_dir
 
 echo $CONTROL > $tmp_dir/tarX
-( cd $pkg_dir && tar $ogargs -czf $tmp_dir/data.tar.gz . -X $tmp_dir/tarX )
-( cd $pkg_dir/$CONTROL && tar $ogargs -czf $tmp_dir/control.tar.gz . )
+( cd $pkg_dir && $TAR $ogargs -X $tmp_dir/tarX -czf $tmp_dir/data.tar.gz . )
+( cd $pkg_dir/$CONTROL && $TAR $ogargs -czf $tmp_dir/control.tar.gz . )
 rm $tmp_dir/tarX
 
 echo "2.0" > $tmp_dir/debian-binary
@@ -254,7 +254,7 @@
 if [ "$outer" = "ar" ] ; then
   ( cd $tmp_dir && ar -crf $pkg_file ./debian-binary ./data.tar.gz ./control.tar.gz )
 else
-  ( cd $tmp_dir && tar -zcf $pkg_file ./debian-binary ./data.tar.gz ./control.tar.gz )
+  ( cd $tmp_dir && $TAR -zcf $pkg_file ./debian-binary ./data.tar.gz ./control.tar.gz )
 fi
 
 rm $tmp_dir/debian-binary $tmp_dir/data.tar.gz $tmp_dir/control.tar.gz
